### PR TITLE
chore: Add support docs for Swift Package Manager

### PR DESCRIPTION
**Pull request by palpatim**
*Description of changes:* Add in-repo support docs for SPM support.

*Check points:*

- [ ] ~~Added new tests to cover change, if needed~~
- [ ] ~~All unit tests pass~~
- [ ] ~~All integration tests pass~~
- [X] Updated CHANGELOG.md
- [X] Documentation update for the change if required
- [X] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README-spm-support.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+The AWS Mobile SDK for iOS can be installed via Swift Package Manager, but you will have to import the target dependencies for each SDK you want to use.
+
+Most SDKs only require you to import the SDK itself, and **AWSCore**:
+
+![Select dependencies](readme-images/spm-dependencies-select.png)
+
+However, some SDKs rely on other libraries in the SDK. The full list is below
+
+| SDK | Dependencies | Notes |
```
Comment (royjit) [inline README-spm-support.md:11]: Thanks for adding this!

File: README.md [hunk 2]
```diff
@@ -37,24 +37,56 @@ To use the AWS SDK for iOS, you will need the following installed on your develo
...
+
+    ![Dependency version options](readme-images/spm-setup-03-dependency-version-options.png)
+
+    **NOTE:** The AWS Mobile SDK for iOS does [not use Semantic Versioning](https://docs.amplify.aws/sdk/configuration/setup-options/q/platform/ios#aws-sdk-version-vs-semantic-versioning), and may introduce breaking API changes on minor version releases. We recommend setting your **Version** rule to **Up to Next Minor** and evaluating minor version releases to ensure they are compatible with your app.
+
+1. Choose which of the libraries you want added to your project. Always select the **AWSCore** SDK. The remaining SDKs to install will vary based on which SDK you're trying to install. Most SDKs rely only on **AWSCore**, but for a full dependency list, see the [README-spm-support file](README-spm-support.md).
+
+    _Note: AWSLex is not currently supported through Swift Package Manager due to conflicts with a packaged binary dependency._
```
Comment (royjit) [inline README.md:83]: AWSLex is supported via SPM, but it is not supported for arm64 - https://github.com/aws-amplify/aws-sdk-ios/pull/3365

AWSLex checksum - https://github.com/aws-amplify/aws-sdk-ios-spm/blob/9057509ddfc1f3ef1c36264e9b8e8173eda634c9/Package.swift#L43

File: README-spm-support.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+| **AWSFacebookSignIn** | AWSCore, AWSAuthCore, [FBSDKLoginKit from Facebook SDK](https://github.com/facebook/facebook-ios-sdk/) | Only needed if you are using Facebook as an Identity Provider for Cognito Federated Identities (Cognito Identity Pools).
+| **AWSGoogleSignIn** | AWSCore, AWSAuthCore | Only needed if you are using Google as an Identity Provider for Cognito Federated Identities (Cognito Identity Pools)
+| **AWSIoT** | AWSCore |
+| **AWSKMS** | AWSCore |
+| **AWSKinesis** | AWSCore |
+| **AWSKinesisVideo** | AWSCore |
+| **AWSKinesisVideoArchivedMedia** | AWSCore |
+| **AWSKinesisVideoSignaling** | AWSCore |
+| **AWSLambda** | AWSCore |
+| **AWSLex** | _SPM not supported due to an incompatibility with a packaged binary_ |
```
Comment (royjit) [inline README-spm-support.md:39]: It is supported

File: README.md (hunk 2 above)
Comment (palpatim) [reply README.md:83]: Thanks for clarifying, will fix

File: README-spm-support.md (hunk 3 above)
Comment (palpatim) [reply README-spm-support.md]: Fixed

**Review by royjit — APPROVED** [review]
LGTM

**Pull request merged**